network/hnswrapper/hnsv2wrapperfake.go

```diff
@@ -167,7 +167,7 @@ func (f Hnsv2wrapperFake) GetNetworkByName(networkName string) (*hcn.HostCompute
 	if network, ok := f.Cache.networks[networkName]; ok {
 		return network.GetHCNObj(), nil
 	}
-	return &hcn.HostComputeNetwork{}, nil
+	return nil, hcn.NetworkNotFoundError{}
```

Comment:
just make sure cni code doesn't dereference pointer without checking for error

Reply:
this is the hns mock wrapper and only used by test code, know real cni code uses this type